Stratis.Bitcoin/BlockPulling/LookaheadBlockPuller.cs

```diff
@@ -403,14 +407,21 @@ private Block NextBlockCore(CancellationToken cancellationToken)
                     }
                     else
                     {
+                        // TODO: Race condition here - as soon as IsDownloading returns false here, 
+                        // another thread can change that (consider BlockPullerBehavior.Node_MessageReceived, 
+                        // just before the newly downloaded block is pushed, the above check for DownloadedBlocks 
+                        // will fail - block not pushed yet - but then IsDownloading here returns false), which 
+                        // will cause this thread to call AskBlocks for a block that was already downloaded. 
+                        // https://github.com/stratisproject/StratisBitcoinFullNode/issues/244
```

Comment:
I've separated this to its own issue, so it should not be considered in this PR, but the case is that one thread executes TryRemoveDownloadedBlock and that returns false (block not ready) and then another thread delivers the block (block ready and no longer is downloading), then the first one executes IsDownloading and it returns false and then we ask for the block that we already have (but is not in the map anymore).

The second part - asking for block which is in map won't throw as this is currently job of AssignDownloadTaskToPeer which uses TryAdd and ignores the request if the block is there in the map.